src/link/script.y

```diff
@@ -460,28 +528,37 @@ static void placeSection(std::string const &name) {
 	section->bank = bank;
 
-	uint16_t &org = curAddr[activeType][activeBankIdx];
-	if (section->isAddressFixed && org != section->org) {
-		scriptError(context, "The linker script assigns section \"%s\" to address $%04" PRIx16 ", but it was already at $%04" PRIx16,
-		            name.c_str(), org, section->org);
-	} else if (section->isAlignFixed && (org & section->alignMask) != section->alignOfs) {
-		uint8_t alignment = std::countr_one(section->alignMask);
-		scriptError(context, "The linker script assigns section \"%s\" to address $%04" PRIx16 ", but that would be ALIGN[%" PRIu8 ", %" PRIu16 "] instead of the requested ALIGN[%" PRIu8 ", %" PRIu16 "]",
-		            name.c_str(), org, alignment, (uint16_t)(org & section->alignMask), alignment, section->alignOfs);
-	}
-	section->isAddressFixed = true;
-	section->isAlignFixed = false; // This can't be set when the above is.
-	section->org = org;
-
-	uint16_t curOfs = org - typeInfo.startAddr;
-	if (section->size > typeInfo.size - curOfs) {
-		scriptError(context, "The linker script assigns section \"%s\" to address $%04" PRIx16 ", but then it would overflow %s by %" PRIx16 " bytes",
-		            name.c_str(), org, typeInfo.name.c_str(),
-		            (uint16_t)(section->size - (typeInfo.size - curOfs)));
-		// Fill as much as possible without going out of bounds.
-		org = typeInfo.startAddr + typeInfo.size;
+	if (!isPcFloating) {
+		uint16_t &org = curAddr[activeType][activeBankIdx];
+		if (section->isAddressFixed && org != section->org) {
+			scriptError(context, "The linker script assigns section \"%s\" to address $%04" PRIx16 ", but it was already at $%04" PRIx16,
+			            name.c_str(), org, section->org);
+		} else if (section->isAlignFixed && (org & section->alignMask) != section->alignOfs) {
+			uint8_t alignment = std::countr_one(section->alignMask);
+			scriptError(context, "The linker script assigns section \"%s\" to address $%04" PRIx16 ", but that would be ALIGN[%" PRIu8 ", %" PRIu16 "] instead of the requested ALIGN[%" PRIu8 ", %" PRIu16 "]",
+			            name.c_str(), org, alignment, (uint16_t)(org & section->alignMask), alignment, section->alignOfs);
+		}
+		section->isAddressFixed = true;
+		section->isAlignFixed = false; // This can't be set when the above is.
+		section->org = org;
+
+		uint16_t curOfs = org - typeInfo.startAddr;
+		if (section->size > typeInfo.size - curOfs) {
+			scriptError(context, "The linker script assigns section \"%s\" to address $%04" PRIx16 ", but then it would overflow %s by %" PRIx16 " bytes",
+			            name.c_str(), org, typeInfo.name.c_str(),
+			            (uint16_t)(section->size - (typeInfo.size - curOfs)));
+			// Fill as much as possible without going out of bounds.
+			org = typeInfo.startAddr + typeInfo.size;
+		} else {
+			org += section->size;
+		}
 	} else {
-		org += section->size;
+		section->isAddressFixed = false;
+		section->isAlignFixed = floatingAlignMask != 1;
+		section->alignMask = floatingAlignMask;
```

Comment:
On the other hand a section's `alignMask` field really *is* a mask; it gets `&`ed, like in `(uint16_t)(org & section->alignMask)` elsewhere here. So I think this should be `section->alignMask = floatingAlignSize - 1;`.